packages/volto/src/components/manage/Blocks/Search/SearchBlockView.jsx

```diff
@@ -66,11 +68,29 @@ const applyDefaults = (data, root) => {
       ? { sort_order: 'descending' }
       : {};
 
+  // We start with the base query from the block.
+  // We enhance it with the query from the facets (filters).
+  // We fall back to the default query.
+  let query = block_query;
+  if (!query.length) {
+    query = data_query.length ? data_query : defaultQuery;
+  } else if (data_query.length) {
+    // We have both a base query and a filter.  Combine them.
+    // Items in the filter win over items in the base query.
+    const filter_keys = data_query.map((obj) => obj.i);
```

Comment:
Use Set for best performance:

```suggestion
    const filterKeys = new Set(dataQuery.map((obj) => obj.i));
```